core/db/migrate/20250604072105_add_fk_products_variant_property_rules.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddFkProductsVariantPropertyRules < ActiveRecord::Migration[7.0]
+  def change
+    add_foreign_key :spree_variant_property_rules, :spree_products, column: :product_id, null: false
```

Comment:
Could this change break migrations on databases with inconsistencies?

Reply:
Edit: I've added error handling that instructs the user to uncomment code that would remove the inconsistent records, giving them the chance to review the removals before running the migrations.